javav2/example_code/ec2/src/main/java/com/example/ec2/Route53Scenario.java

```diff
@@ -0,0 +1,394 @@
+
+        System.out.println(DASHES);
+        System.out.println("7. Check domain transferability.");
+        checkDomainTransferability(route53DomainsClient, domainSuggestion);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("8. Request a domain registration.");
+        String opId = requestDomainRegistration(route53DomainsClient, domainSuggestion);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("9. Get an operation detail.");
+        getOperationalDetail(route53DomainsClient, opId);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("10. Get domain details.");
+        System.out.println("Note: you must have a registered domain to get details.");
+        System.out.println("Otherwise an exception is thrown that states " );
+        System.out.println("Domain xxxxxxx not found in xxxxxxx account.");
+        getDomainDetails(route53DomainsClient, domainSuggestion);
+        System.out.println(DASHES);
+
+    }
+
+    public static void getDomainDetails(Route53DomainsClient route53DomainsClient, String domainSuggestion){
+        try {
+            GetDomainDetailRequest detailRequest = GetDomainDetailRequest.builder()
+                .domainName(domainSuggestion)
+                .build();
+
+            GetDomainDetailResponse response = route53DomainsClient.getDomainDetail(detailRequest);
+            System.out.println("The contact first name is " + response.registrantContact().firstName());
+            System.out.println("The contact last name is " + response.registrantContact().lastName());
+            System.out.println("The contact org name is " + response.registrantContact().organizationName());
+
+        } catch (Route53Exception e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+
+    }
+
+    public static void getOperationalDetail(Route53DomainsClient route53DomainsClient, String operationId) {
+        try {
+            GetOperationDetailRequest detailRequest = GetOperationDetailRequest.builder()
+                .operationId(operationId)
+                .build();
+
+            GetOperationDetailResponse response = route53DomainsClient.getOperationDetail(detailRequest);
+            System.out.println("Operation detail message is "+response.message());
+
+        } catch (Route53Exception e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+
+    }
+
+
+    public static String requestDomainRegistration(Route53DomainsClient route53DomainsClient, String domainSuggestion) {
+        try {
+            ContactDetail contactDetail = ContactDetail.builder()
+                .contactType(ContactType.COMPANY)
+                .state("LA")
+                .countryCode(CountryCode.IN)
+                .email("scmacdon@amazon.com")
+                .firstName("Scott")

```

Comment:
Should be a generic email and name here. If not loading from a file, you may also want to pass in the email address because the user may want to use their own.

Reply:
will fix